### DISCUSSION TITLE
How to remove a value in `overrides`

### DISCUSSION THREAD

**Discussion by ambroisie** [ellisonleao/gruvbox.nvim#321] [Q&A] [answered]
I want to use `overrides` to e.g: revert the `DiffAdd` change.

How can I _remove_ the `bg` attribute through an override? The following does _not_ work, because a `nil` value in a table is the same as not having that key in the table...

```lua
gruvbox.setup({
    overrides = {
        DiffAdd = { fg = colors.green, bg = nil },
    },
})
```

EDIT: this is with #319 applied, to be able to refer to the `colors` palette properly.

**Comment by ambroisie** [comment]
BTW, `bg = colors.bg0` does not solve the issue, since it will not mesh well with what is _under_ the diff colour (e.g: `colorcolumn`):

Here it is with `bg = colors.bg0`:
![](https://github.com/ellisonleao/gruvbox.nvim/assets/12465195/3e71d812-aa68-49b9-af5f-3b744ee34b59)

And here is what it should look like with a proper revert:
![image](https://github.com/ellisonleao/gruvbox.nvim/assets/12465195/b708bb7c-82c5-4e09-b60b-863d67a99417)

**Comment by ambroisie** [comment]
IMO, the solution would be:

1. Apply #319 to expose the color palette in a way that is usable for `overrides`.
2. Do _not_ use `vim.tbl_extend("force", ...)` when applying the `overrides`, but just _assign_ the new value to it (so no merging)
    * Thanks to 1., a user can make use of `colors` to properly refer to the palette colours, no matter if they're using low/high contrast. So the need for _merging_ is removed.

**Comment by darkangel-ua** [comment, marked as answer]
Maybe just 
```
gruvbox.setup({
    overrides = {
        DiffAdd = { fg = colors.green, bg = "NONE" },
    },
})
```
?
